File: frontend/src/types/report.ts
export type ReportStatus = 'pending' | 'done' | 'error'

export interface WeeklyReport {
  id: number
  user_id: number
  week_start: string   // YYYY-MM-DD (Monday)
  status: ReportStatus
  content: string | null
  error_msg: string | null
  created_at: string
  updated_at: string
}

File: backend/app/schemas/report.py
from datetime import date, datetime

from pydantic import BaseModel


class WeeklyReportResponse(BaseModel):
    id: int
    user_id: int
    week_start: date
    status: str
    content: str | None
    error_msg: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
